src/evaluation_system/api/parameters.py

```diff
@@ -1154,9 +1191,12 @@ def parse(self, value: Any) -> str:
         ValueError:
             if value is not part of possible options.
         """
-
-        if self._verified(value):
-            for key, val in self.options.items():
-                if value == val:
-                    return key
-        return ""
+        values = self._verified(value)
+        if self.multiple is False and len(values) > 1:
+            raise ValidationError(
+                f'Multiple values are not allowed for "{self.name}". '
+                f"Only a single value is permitted."
+            )
+        if len(values) > 1:
+            self.max_items = len(values)
```

Comment:
What happens if we don't set the `max_items` attribute here and instead left it at `sys.maxsize`, as defined in the init? Asking to better understand how the parser works / is used in the evaluation system workflow.

Reply:
Thanks for the suggestion, I did have a look at the code angain and spotted a flaw. I'll change that.